frontend/__tests__/services/cct-service.server.test.ts

```diff
@@ -0,0 +1,83 @@
+import { http } from 'msw';
+import { setupServer } from 'msw/node';
+import * as fs from 'node:fs';
+import path from 'path';
+import { afterAll, afterEach, beforeAll, describe, expect, expectTypeOf, it, vi } from 'vitest';
+
+import { getPdfEntity } from '~/mocks/cct-api.server';
+import { getCCTService } from '~/services/cct-service.server';
+
+const handlers = [
+  http.get('https://api.example.com/cct/letters/:referenceId', async ({ params, request }) => {
+    const pdfEntity = getPdfEntity(params.referenceId);
+    const filePath = path.join(__dirname, '..', '..', 'public', 'test.pdf');
+
+    try {
+      const stats = await fs.promises.stat(filePath);
+      const nodeStream = fs.createReadStream(filePath);
+      const stream = new ReadableStream({
+        start(controller) {
+          nodeStream.on('data', (chunk) => {
+            controller.enqueue(chunk);
+          });
+          nodeStream.on('end', () => {
+            controller.close();
+          });
+          nodeStream.on('error', (err) => {
+            controller.error(err);
+          });
+        },
+      });
+
+      return new Response(stream, {
+        headers: {
+          'Content-Type': 'application/pdf',
+          'Content-Length': stats.size.toString(),
+          'Content-Disposition': `inline; filename="${pdfEntity.id}.pdf"`,
+        },
+      });
+    } catch (error) {
+      return new Response('Failed to fetch PDF', { status: 500 });
+    }
```

Comment:
**Nitpick:** I don't think you needed to implement all of this for the mock. You could have just returned a short string.

The current implementation is a bit fragile because it relies on an external PDF file which might not always exist.